### PR TITLE
add anonymous access policy control

### LINKED ISSUE #4762: Registry: add unauthenticated option to project policy
With https://github.com/openshift/origin/pull/9887 a project can be created to allow unauthenticated users to pull images. An option should be added to the "new project" dialog to support this. 

> Allow any unauthenticated user to pull images

How can we differentiate between this and authenticated users? We need to make it clear.

Technically we need call the API with this:

`oc policy add-role-to-user registry-viewer system:anonymous -n <project>`

cc @thomasmckay

### PR DESCRIPTION
**Pull request by aweiteka**
Resolves #4762

### CODE REVIEW

**Comment by stefwalter** [comment]
There's a semicolon missing:

```
../pkg/kubernetes/scripts/projects.js:417:41: Missing semicolon.
```

**Comment by stefwalter** [comment]
@aweiteka So there's more work here. Do you want me to finish this up? If so, can you summarize the intent and why a simple fix doesn't suffice?

**Comment by aweiteka** [comment]
> Do you want me to finish this up?

Yes, I haven't gotten back to this. if you would that would be great.

> can you summarize the intent and why a simple fix doesn't suffice?

I pushed a commit that should describe what's needed. When updating a project to "Private" we need to ensure we remove both system:authenticated and system:unauthenticated that could be assigned. I wasn't clear how to make multiple calls to `policy.removeFromRole()`.

Use `oc get rolebindings` to view groups assigned **registry-viewer**.

**Comment by andreasn** [comment]
poicy -> policy
for the git log message I think

**Comment by aweiteka** [comment]
@stefwalter should be all set now, added test, squashed. This has high visibility so I'd like to get this in ASAP.

**Comment by petervo** [comment]
Some missing ;'s

```
pkg/kubernetes/scripts/projects.js:432:18: Missing semicolon.
pkg/kubernetes/scripts/projects.js:436:18: Missing semicolon.
pkg/kubernetes/scripts/projects.js:983:23: Missing semicolon.
```

**Comment by stefwalter** [comment]
This needs a new build of the openshift test VM to actually verify that this works. I'm building that.

**Comment by stefwalter** [comment]
@aweiteka I can't get this to work with an updated openshift instance.

```
# sudo docker pull registry:5000/marmalade/busybee:0.x
Trying to pull repository registry:5000/marmalade/busybee ... 
Pulling repository registry:5000/marmalade/busybee
Error: image marmalade/busybee not found
```

Is this functionality not part of the openshift container yet? Do you know how I would find out?

**Comment by stefwalter** [comment]
Replaced by #4848
